include/seqan3/range/view/detail.hpp

```diff
@@ -139,38 +132,37 @@ struct declare_view_functor_type
      * // in both cases v is now of type view_foo<std::vector<int>>
      * ```
      */
-    template <typename urng_t,
-              typename ...arg_types>
-    auto operator()(urng_t && urange,
-                    arg_types && ...args) const
-        requires requires { view_type{std::declval<urng_t>(), std::declval<arg_types>()...}; }
+    template <typename urng_t, typename ...arg_types>
+    auto operator()(urng_t && urange, arg_types && ...args) const
+        requires input_range_concept<urng_t>
     {
-        return view_type{std::forward<urng_t>(urange), std::forward<arg_types>(args)...};
+        return static_cast<derived_type const *>(this)->impl(std::forward<urng_t>(urange),
+                                                             std::forward<arg_types>(args)...);
     }
 
-    /*!\brief Binds operator() with a placeholder for range so that the operator() without `urange` argument works
-     * [for view functors with arguments].
-     * \tparam arg_types The types of the arguments.
+    /*!\brief   Binds operator() with a placeholder for range so that the operator() without `urange` argument works
+     *          [for adaptors with arguments].
+     * \tparam  arg_types The types of the arguments.
      *
      * \details
      *
      * This replaces std::bind which doesn't work here because of the parent being template template and thus
      * the type not being fully resolved. A lambda function wrapper could be used, but its type wouldn't be
      * deducible which we need for the definition of the pipe operator.
+     *
+     * TODO re-evaluate this since pipable_adaptor_base is no longer template template
      */
     template <typename ...arg_types>
     struct auxiliary_functor_t
```

Comment:
I disagree, it's a local type definition that is used nowhere and not exposed.

Reply:
I think it really doesn't matter whether it is or it is not exposed to the public.
IMHO `..._t` should be used in two cases: a) as the name of a template parameter in a template class declaration or b) as a shortcut for a type traits metafunctions
In this case we name a very fixed type `..._t`, only because no one sees it? Where is the line? It can cause confusion and should not be adopted in other places. So please rename it. And `pipe_adapter_fn_proxy` would be a very telling name to what this class actually is about. But the actual name really isn't the point.

Reply:
To me `_t` or `_type` indicates "typename used in a local context" which is true for template parameters, local typedefs, but also local type definitions. Also we already use this convention in other places and this PR wouldn't otherwise touch those lines.
If you feel that this is important we should make an issue and decide on a policy for the library, but I don't think that it's overly important for this PR, because we are.talking about old code.

Reply:
Yes, I will open an issue for this and we can discuss this there.
Having a separate commit or a new PR for this little change is IMO not relevant, so we can do it later when fixing other places related to the same issue. 👍